libs/wingc/src/lsp/rename_request.rs

```diff
@@ -0,0 +1,422 @@
+                $new_word,
+                text_edit.new_text,
+                "new_word did not match"
+              );
+              text_edit_ranges.push(text_edit.range.clone());
+            }
+        }
+      }
+
+        assert_eq!(
+            get_ranges($code),
+						text_edit_ranges,
+						"target_range did not match"
+					);
+
+				$($assertion)*
+			}
+		};
+	}
+
+	test_rename_request!(
+		variable_same_scope,
+		r#"
+let thing = "thing";
+  //-----
+let thing2 = thing;
+           //----^
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		variable_same_scope_declaration,
+		r#"
+let thing = "thing";
+  //----^
+let thing2 = thing;
+           //-----
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_str,
+		r#"
+let thing = "thing";
+               //^
+let thing2 = thing;
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_symbol,
+		r#"
+let thing = "thing";
+                 //^
+let thing2 = thing;
+		"#,
+		"t1"
+	);
+	//
+
+	test_rename_request!(
+		rename_inner_scope,
+		r#"
+let var thing = 6;
+      //-----
+if (thing > 8) {
+  //-----
+  thing = 6;
+//----^
+} else {
+  if (thing < 0) {
+    //-----
+    let thing = 0;
+    log(thing);
+  }
+  log(thing)
+    //-----
+}               
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_outer_scope,
+		r#"
+    let var thing = 6;
+          //----^
+    if (thing > 8) {
+      //-----
+    thing = 6;
+  //-----
+    } else {
+    if (thing < 0) {
+      //-----
+      let thing = 0;
+      log(thing);
+    }
+    log(thing)
+      //-----
+    }
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_inner_scope_shadowing,
+		r#"
+    let var thing = 6;
+    if (thing > 8) {
+    thing = 6;
+
+    } else {
+    if (thing < 0) {
+      let thing = 0;
+        //-----
+      log(thing);
+        //----^
+    }
+    log(thing)
+    }
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_class,
+		r#"
+class User {
+    //----
+  name: str;
+}
+
+class Admin extends User {
+                  //----
+  permission: num;
+}
+
+let a = new User();
+          //---^
+let b = new Admin();
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_closure,
+		r#"
+let pow = inflight (x: num) => {
+  //--^
+  return x * x;
+};
+
+test "pow" {
+  assert(pow(5) == 25);
+       //---
+  assert(pow(25) == pow( pow(5) ));
+       //---      //---//---
+}
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_for_iterator,
+		r#"
+    bring cloud;
+
+    let buckets = [new cloud.Bucket()];
+    for bucket in buckets {
+      //-----^
+      bucket.addObject("bucket","bucket");
+    //------
+      log(bucket.toString());
+        //------
+    }
+		"#,
+		"t1"
+	);
+
+	test_rename_request!(
+		rename_for_iterator_reference,
+		r#"
+    bring cloud;
+
+    let buckets = [new cloud.Bucket()];
+    for bucket in buckets {
+      //------
+      bucket.addObject("bucket","bucket");
+    //------
+      log(bucket.toString());
+        //-----^
+    }
+		"#,
+		"t1"
```

Comment:
Noticed this is the same in every test so it could be removed and just use a default value for now

Reply:
we're still checking it during the test :) 
I can of course change it to a constant within the test function, but as soon as we fix the namespace case- we'll have to add it again for the tests